hpx/lcos/detail/future_data.hpp

```diff
@@ -333,14 +348,31 @@ namespace detail
           : base_type(no_addref)
         {}
 
+        struct default_construct {};
+
+        future_data_base(init_no_addref no_addref, default_construct)
+          : base_type(no_addref)
+        {
+            result_type* value_ptr = reinterpret_cast<result_type*>(&storage_);
+            construct(value_ptr);
+            state_ = value;
+        }
+
         template <typename Target>
```

Comment:
Is this constructor still needed?

Reply:
The constructor is not needed anymore and has been removed.